CMakeLists.txt

```
@@ -56,6 +56,14 @@ check_c_compiler_flag(-Wno-unused-parameter UV_LINT_NO_UNUSED_PARAMETER)
 check_c_compiler_flag(-Wstrict-prototypes UV_LINT_STRICT_PROTOTYPES)
 check_c_compiler_flag(-Wextra UV_LINT_EXTRA)
 
+string(CONCAT msvc-utf8 $<
+  $<AND:
+    $<C_COMPILER_ID:MSVC>,
+    $<VERSION_GREATER:${MSVC_VERSION},1800>
+  >:/source-charset:utf-8
```

```suggestion
  >:/utf-8
```

How does this compare against https://github.com/libuv/libuv/pull/2889? I'm thinking merge this and close that?

---

^ @slurps-mad-rips I don't have this environment setup, so I wanted to know if this makes sense to you

---

hi sorry. I've been busy with personal life. The /utf-8 will set the execution charset to utf-8 as well. As long as this is desired, it's totally fine. :)

---

Probably makes more sense than having MSVC rewrite the strings that it puts into the binary into whatever happens to be the current code page on the system during compilation. Thanks!